src/org/traccar/protocol/WondexFrameDecoder.java

```diff
@@ -36,17 +38,17 @@ protected Object decode(
         if (buf.getUnsignedByte(buf.readerIndex()) == 0xD0) {
 
             // Send response
-            ChannelBuffer frame = buf.readBytes(KEEP_ALIVE_LENGTH);
-            if (channel != null) {
-                channel.write(frame);
+            ByteBuf frame = buf.readBytes(KEEP_ALIVE_LENGTH);
+            if (ctx != null && ctx.channel() != null) {
```

Comment:
You have channel here as a separate variable, so I don't think we need to use context.